File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int XOR1=0;
        int XOR2=0;
          for(int i:nums){
            XOR1=XOR1^i;
          }
        for(int i=0;i<=nums.size();i++){
            XOR2=XOR2^i;
        }
        return XOR1^XOR2;
    }
};

File: 0268-missing-number/0268-missing-number.java
class Solution {
    void swap(int[] nums,int i,int j){
        int temp=nums[i];
        nums[i]=nums[j];
        nums[j]=temp;
    }
    void cyclic(int[] arr){
    int start=0;
    while(start<arr.length){
        int curr=arr[start];
        if(arr[start]<arr.length&&arr[start]!=arr[curr]){
            swap(arr,start,curr);
        }
        else{
            start++;
        }
    }

    }
    public int missingNumber(int[] nums) {
        cyclic(nums);
        for(int i=0;i<nums.length;i++){
            if(nums[i]!=i){
                return i;
            }
        }
        return nums.length;

        
    }
}